torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -69,14 +69,14 @@ def parallelize_llama(
                 "fused_rmsnorm is not compatible with torch.compile yet. "
                 "Please use rmsnorm or layernorm."
             )
-        model = apply_compile(model)
+        apply_compile(model)
 
     if parallel_dims.dp_enabled:
         if parallel_dims.dp_type == "fsdp":
             dp_mesh = world_mesh["dp"] if world_mesh.ndim > 1 else world_mesh
             assert dp_mesh.mesh_dim_names == ("dp",), dp_mesh.mesh_dim_names
 
-            model = apply_fsdp(
+            apply_fsdp(
```

Comment:
does this change break anything?  IIRC one feature required us to keep returning the model.  Maybe it was AC?

Reply:
> one feature required us to keep returning the model.

Great point. Yes torch.compile and AC require reassigning the model. But since we are doing per-block compile and AC, we achieve that in-place for the whole model by
```
transformer_block = compile/AC(transformer_block)
model.layers.register_module(layer_id, transformer_block)
```

Reply:
any motivation of removing the assignment? I thought an explicit assignment does not look bad

Reply:
@wanchaol 
I actually don't know why these functions have redundant semantics, i.e. modifying a module in-place, but at the same time return it explicitly. I'm modifying it because:
1. Before this PR, the PP branch explicitly reassign the returned module, but the SPMD branch doesn't. I think we should use the minimum viable code to reduce confusion.
2. IIRC @awgu had a PR which removes the reassigning for FSDP2 `fully_shard`, so in some sense I'm mimicking that PR.